Explorer/Assets/Scripts/Global/Dynamic/PortableExperiences/PortableExperiencesController.cs

```diff
@@ -0,0 +1,218 @@
+﻿using Arch.Core;
+using CommunicationData.URLHelpers;
+using Cysharp.Threading.Tasks;
+using DCL.Diagnostics;
+using DCL.Ipfs;
+using DCL.Utilities;
+using DCL.Web3.Identities;
+using DCL.WebRequests;
+using ECS;
+using ECS.SceneLifeCycle;
+using ECS.SceneLifeCycle.SceneDefinition;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using DCL.Utilities.Extensions;
+using SceneRunner.Scene;
+using SceneRuntime.Apis.Modules.PortableExperiencesApi;
+using System.Linq;
+using GetSceneDefinition = ECS.SceneLifeCycle.SceneDefinition.GetSceneDefinition;
+
+namespace PortableExperiences.Controller
+{
+    public class PortableExperiencesController : IPortableExperiencesController
+    {
+        private static readonly QueryDescription GET_SCENE_DEFINITION = new QueryDescription().WithAll<GetSceneDefinition>();
+        private static readonly QueryDescription SCENE_DEFINITION_COMPONENT = new QueryDescription().WithAll<SceneDefinitionComponent>();
+
+        private readonly ServerAbout serverAbout = new ();
+        private readonly IWeb3IdentityCache web3IdentityCache;
+        private readonly IWebRequestController webRequestController;
+        private readonly IScenesCache scenesCache;
+
+        private readonly RealmData realmData = new ();
+
+        private readonly ObjectProxy<World> globalWorldProxy;
+        private List<Entity> entitiesToDestroy = new ();
+        public Dictionary<string, Entity> PortableExperienceEntities { get; } = new ();
+        private World world => globalWorldProxy.Object;
+        private List<IPortableExperiencesApi.SpawnResponse> spawnResponsesList = new ();
+
+        public PortableExperiencesController(
+            ObjectProxy<World> world,
+            IWeb3IdentityCache web3IdentityCache,
+            IWebRequestController webRequestController,
+            IScenesCache scenesCache)
+        {
+            globalWorldProxy = world;
+            this.web3IdentityCache = web3IdentityCache;
+            this.webRequestController = webRequestController;
+            this.scenesCache = scenesCache;
+        }
+
+        public async UniTask<IPortableExperiencesApi.SpawnResponse> CreatePortableExperienceAsync(string ens, string urn, CancellationToken ct, bool isGlobalPortableExperience = false)
```

Comment:
I think this method could be simplified if you just pass the `aboutUrl` as parameter, deleting the `urn` param

Reply:
As far as I have seen in Kernel code, a PX can be loaded from a scene using either an ens or urn, if its ens, we get the about url address by adding it the world url https://worlds-content-server.decentraland.org/world/ and then the /about.
IF its an urn, apparently we must use the current realm data to load the scene. I have not implemented that part yet as it seems a bit fishy and tbh, its not needed for current shape.
So if they pass the urn, I cannot just use the aboutUrl as I still need the scene urn to load it and it wont be in the list of scenes from the realmData...

Reply:
Then perhaps we should have two different overloads? One by ens name, the other by url?

Reply:
Yep good idea :)